File: subagentic_ai_py/subagentic_ai/security_level.py
"""Security Level (SL) enumeration for outbound information flow control."""

from enum import Enum


class SecurityLevel(Enum):
    """
    Security Level (SL) governs outbound information flow.
    Specifies the conditions under which a SubAgent may disclose information to external entities.
    """
    
    PUBLIC = 0  # Public information - no restrictions on disclosure
    INTERNAL = 1  # Internal use only - restricted to organization
    CONFIDENTIAL = 2  # Confidential - limited disclosure
    SECRET = 3  # Secret - highly restricted disclosure
    TOP_SECRET = 4  # Top Secret - maximum restrictions

File: SubAgenticAI.Core/SecurityLevel.cs
namespace SubAgenticAI.Core;

/// <summary>
/// Security Level (SL) governs outbound information flow.
/// Specifies the conditions under which a SubAgent may disclose information to external entities.
/// </summary>
public enum SecurityLevel
{
    /// <summary>
    /// Public information - no restrictions on disclosure
    /// </summary>
    Public = 0,
    
    /// <summary>
    /// Internal use only - restricted to organization
    /// </summary>
    Internal = 1,
    
    /// <summary>
    /// Confidential - limited disclosure
    /// </summary>
    Confidential = 2,
    
    /// <summary>
    /// Secret - highly restricted disclosure
    /// </summary>
    Secret = 3,
    
    /// <summary>
    /// Top Secret - maximum restrictions
    /// </summary>
    TopSecret = 4
}
